### PR TITLE
Fix OpenMP_FOUND flag, fix macOS OpenMP include path

### PR DESCRIPTION
**Pull request by vespakoen**
Fixes issue https://github.com/solvespace/solvespace/issues/840 and also fixes OpenMP on macOS when it's used in 3rd-party libraries (found this bug when rebasing the Eigen PR).

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -118,6 +118,16 @@ if(ENABLE_OPENMP)
     find_package( OpenMP REQUIRED )
     include(FindOpenMP)
     if(OPENMP_FOUND)
```
Comment (ruevs) [inline CMakeLists.txt:120]: Here and on line 130?

File: CMakeLists.txt [hunk 2]
```diff
@@ -118,6 +118,16 @@ if(ENABLE_OPENMP)
...
+    if (APPLE)
+        set(OpenMP_CXX_FLAGS "-Xpreprocessor -fopenmp")
+        set(OpenMP_CXX_LIB_NAMES "omp")
+        set(OpenMP_omp_LIBRARY "/usr/local/opt/libomp/lib/libomp.dylib")
+        include_directories("/usr/local/opt/libomp/include")
+    else()
+        find_package( OpenMP REQUIRED )
+        include(FindOpenMP)
+    endif()
+    if(OpenMP_FOUND)
```
Comment (ruevs) [inline CMakeLists.txt:130]: ?

**Comment by vespakoen** [comment]
Whoops, bad copy paste job, updated the PR.

**Comment by vespakoen** [comment]
This is not correct, the macOS stuff doesn't make sense, closing it until I find the solution.

**Pull request closed without merging**